### ISSUE TITLE
False positive for MA0071 (Avoid using redundant else) rule in try/catch

### ISSUE THREAD

**Issue by louis-z** [meziantou/Meziantou.Analyzer#179]
With the following code, the `else` statement is flagged as redundant, but it shouldn't be:
```cs
try
{
    DoSomething();
}
catch (Exception ex)
{
    if (ex is InvalidOperationException )
    {
        // Handle the exception
    }
    else
    {
        throw;
    }
}
```

**Comment by meziantou** [comment]
Thanks for reporting the issue!

Do you want to fix it?

**Comment by louis-z** [comment]
Sure, but I may not have time to so so anytime soon.

**Comment by meziantou** [comment]
I think this will require more info to reproduce the bug. The following test doesn't reproduce the issue:

````
[Fact]
public async Task Test_EmptyIf()
{
    var originalCode = @"
using System;
class TestClass
{
void Test()
{
    try
    {
        //DoSomething();
    }
    catch (Exception ex)
    {
        if (ex is ArgumentException)
        {
            // test
        }
        else // 👈 No report here
        {
            throw;
        }
    }
}
}";
    await CreateProjectBuilder()
            .WithSourceCode(originalCode)
            .ValidateAsync();
}
````

**Comment by louis-z** [comment]
So I tried reproducing the bug, but to no avail. (I must have left out important details when I tried to summarize the problem 😞) 
Issue should be closed.
